src/test/java/com/ge/predix/uaa/token/lib/FastTokenServiceTest.java

```diff
@@ -1,303 +0,0 @@
-        String accessToken = this.testTokenUtil.mockAccessToken(60);
-
-        // Start tamper ;)
-        String[] jwtParts = accessToken.split("\\.");
-        String jwtHeader = jwtParts[0];
-        String jwtContent = jwtParts[1];
-        String jwtSignature = jwtParts[2];
-
-        ObjectMapper objectMapper = new ObjectMapper();
-        TypeReference<Map<String, Object>> valueTypeRef = new TypeReference<Map<String, Object>>() {
-            // Nothing to declare.
-        };
-        String decodedClaims = new String(Base64.decodeBase64(jwtContent));
-        Map<String, Object> claims = objectMapper.readValue(decodedClaims, valueTypeRef);
-        claims.put(USER_ID, "admin");
-        String encodedClaims = Base64.encodeBase64String(objectMapper.writeValueAsBytes(claims));
-        accessToken = jwtHeader + "." + encodedClaims + "." + jwtSignature;
-
-        // We've tampered the token so this should fail.
-        this.services.loadAuthentication(accessToken);
-    }
-
-
-
-    /**
-     * Tests that connection error while retrieving token key issues RestClientException.
-     */
-    @SuppressWarnings("unchecked")
-    @Test(expectedExceptions = RestClientException.class)
-    public void testLoadAuthenticationWithConnectionTimeout() throws Exception {
-        String accessToken = this.testTokenUtil.mockAccessToken(60);
-
-        FastTokenServices services = new FastTokenServices();
-        services.setTrustedIssuers(trustedIssuers());
-        services.afterPropertiesSet();
-        ParameterizedTypeReference<Map<String, Object>> typeRef =
-                new ParameterizedTypeReference<Map<String, Object>>() {
-            // Nothing to add.
-        };
-        RestTemplate restTemplate = Mockito.mock(RestTemplate.class);
-        Mockito.when(restTemplate.exchange(TOKEN_KEY_URL, HttpMethod.GET, null, typeRef))
-                .thenThrow(RestClientException.class);
-        services.setRestTemplate(restTemplate);
-
-        services.loadAuthentication(accessToken);
-    }
-
-    /**
-     * This tests that we can extract the issuer from the token claims.
-     */
-    @Test
-    public void testGetIssuerFromClaims() {
-        String accessToken = this.testTokenUtil.mockAccessToken(60);
-
-        assertEquals(TOKEN_ISSUER_ID, this.services.getIssuerFromClaims(this.services.getTokenClaims(accessToken)));
-
-    }
-
-    /**
-     * This tests that we can derive the token_key endpoint from the issuer id. E.g.
-     * http://localhost:8080/uaa/oauth/token -> https://localhost:8080/uaa/token_key
-     */
-    @Test
-    public void testGetTokenKeyURL() {
-        assertEquals("https://localhost:8080/uaa/token_key", this.services.getTokenKeyURL(TOKEN_ISSUER_ID));
-
-        assertEquals("https://sample.com/token_key",
-                this.services.getTokenKeyURL("https://sample.com/oauth/token"));
-    }
-
-    @Test(expectedExceptions = InvalidTokenException.class)
-    public void testVerifyTimeWindowException() {
```

Comment:
we are not adding this test just because previous manual verification of token expiry logic removed?

Reply:
Added